docarray/array/mixins/evaluation.py

```diff
@@ -1,59 +1,147 @@
 import warnings
-from typing import Optional, Union, TYPE_CHECKING, Callable
+from typing import Optional, Union, TYPE_CHECKING, Callable, List, Dict
 
 import numpy as np
+from collections import defaultdict
 
 from docarray.score import NamedScore
 
 if TYPE_CHECKING:
     from docarray import Document, DocumentArray
 
 
+def _evaluate_deprecation(f):
+    """Raises a deprecation warning if the user executes the evaluate function with
+    the old interface and adjust the input to fit the new interface."""
+
+    def func(*args, **kwargs):
+        if len(args) > 1:
+            if not (
+                isinstance(args[1], Callable)
+                or isinstance(args[1], str)
+                or isinstance(args[1], list)
+            ):
+                kwargs['ground_truth'] = args[1]
+                args = [args[0]] + list(args[2:])
+                warnings.warn(
+                    'The `other` attribute in `evaluate()` is transfered from a '
+                    'positional attribute into the keyword attribute `ground_truth`.'
+                    'Using it as a positional attribute is deprecated and will be removed '
+                    'in the next version.',
```

Comment:
not the next. say soon

Reply:
I will first change this in https://github.com/jina-ai/docarray/pull/617